### PR TITLE
Remove auth handling endpoints

### PR DESCRIPTION
**Pull request by smcclure17**
with #36 We no longer need REST endpoints to handle API keys

### CODE REVIEW

File: functions/src/auth.ts [hunk 1]
```diff
@@ -39,21 +37,20 @@ export function isAPIKeyAuthorized(
...
-  res: Response,
-  next: NextFunction
-) {
-  const idToken = req.get("Authorization")?.split("Bearer ")[1];
-  verifyIdToken(idToken)
-    .then(() => next())
-    .catch((error) => {
-      sendAndThrowShareLinkOrUnexpectedError(error, res);
-    });
+async function isValidKey(apiKey: string | undefined): Promise<boolean> {
```
Comment (smcclure17) [inline functions/src/auth.ts:45]: Ported this over from the old `ApiKeyHandler` class

File: functions/src/APIKeyHandler.ts [hunk 2]
```diff
@@ -1,88 +0,0 @@
-import { randomUUID } from "crypto";
```
Comment (smcclure17) [inline functions/src/APIKeyHandler.ts:1]: We handle most of this in the Web UI now, so we can remove this

File: functions/src/tests/utils.ts [hunk 3]
```diff
@@ -87,22 +89,15 @@ export async function getOrRegisterIdToken(
 }
 
 /**
- * Retrieves the existing or registers a new API key for the given email.
- * @param email Email to register API key for.
- * @param idToken ID token to use for authentication.
- * @returns The API key.
+ * Inserts a mock API key document into the Firestore database.
```
Comment (mikelehen) [inline functions/src/tests/utils.ts]: Is there something about this that makes it a "mock" key?  It'll be a real key, no?

**Review by mikelehen — APPROVED** [review]
Looks great, thanks for the cleanup!

File: functions/src/tests/utils.ts (hunk 3 above)
Comment (smcclure17) [reply functions/src/tests/utils.ts]: Good point--this isn't really a mock, I've removed that language 🙏

**Pull request merged**